components/ShikiHighlighter.jsx

```diff
@@ -25,6 +30,37 @@ module.exports = class ShikiHighlighter extends React.PureComponent {
     clipboard.writeText(this.props.content)
   }
 
+  openInVSC () {
+    if (this.state.vscodeCooldown) return
+
+    this.setState({
+      vscodeCooldown: true
+    })
+
+    setTimeout(() => {
+      this.setState({
+        vscodeCooldown: false
+      })
+    }, 1000)
+
+    try {
+
+      try {
+        const path = __dirname + '/'
+        let regex = /tempFile/
+        fs.readdirSync(path).filter(f => regex.test(f)).map(f => fs.unlinkSync(path + f))}
+      catch {console.log('No File Present, Skipping.')}
+
+      let language = '.'+this.props.lang
+      if (!this.props.lang) {language = ''}
+
+      fs.writeFileSync(filePath+language, this.props.content);
+      setTimeout(() => {open(vscodeURI+language);}, 500); //making sure the file has been written in time
```

Comment:
Afaik you shouldn't need to wait for the file to write since you are using the synchronous method for writing, which doesn't return until the file is written

Reply:
Resolved. No More waiting!